CloverConnector/Classes/cloverconnector/DefaultCloverDevice.swift

```diff
@@ -34,13 +34,16 @@ class DefaultCloverDevice : CloverDevice, CloverTransportObserver {
         if let transport = config.getTransport() {
             super.init(packageName: config.getMessagePackageName(), transport: transport)
             transport.subscribe(self)
-            transport.initialize()
         } else {
             return nil      
         }
         
     }
-    
+
+    override func initialize() {
+        transport.initialize()
+    }
```

Comment:
this is unrelated to the error handling, but `transport.initialize` was being called from the `DefaultCloverDevice`'s initializer originally, which is not an appropriate place because observers may not have been set up yet.